tests/Feature/EmailAssignmentTest.php

```diff
@@ -0,0 +1,176 @@
+<?php
+
+namespace Tests\Feature;
+
+use Illuminate\Foundation\Testing\DatabaseTransactions;
+use Illuminate\Support\Facades\Notification;
+use Tests\TestCase;
+
+class EmailAssignmentTest extends TestCase
+{
+    use DatabaseTransactions;
+
+    private $account;
+
+    public function setUp()
+    {
+        parent::setUp();
+
+        // fakes notifications for the entire test class
+        Notification::fake();
+
+        $this->account = factory(\App\Models\Mship\Account::class)->create();
+    }
+
+    /** @test * */
+    public function testUserCanAccessEmailAddForm()
+    {
+        $this->actingAs($this->account)->get(route('mship.manage.email.add'))
+            ->assertSuccessful();
+    }
+
+    /** @test * */
+    public function testRedirectOnInvalidEmail()
+    {
+        $data = [
+            'new_email' => 'not_an_email.com',
+            'new_email2' => 'not_an_email.com',
+        ];
+
+        $this->actingAs($this->account)->post(route('mship.manage.email.add.post'), $data)
+            ->assertRedirect(route('mship.manage.email.add'))
+            ->assertSessionHas('error', 'You have entered an invalid email address.');
+    }
+
+    /** @test * */
+    public function testRedirectOnNonMatchingEmail()
+    {
+        $data = [
+            'new_email' => 'matching.email@example.com',
+            'new_email2' => 'not.matching.email@example.com',
+        ];
+
+        $this->actingAs($this->account)->post(route('mship.manage.email.add.post'), $data)
+            ->assertRedirect(route('mship.manage.email.add'))
+            ->assertSessionHas('error', 'Emails entered are different.  You need to enter the same email, twice.');
+    }
+
+    /** @test * */
+    public function testSuccessfulPostEmail()
+    {
+        $data = [
+            'new_email' => 'email@example.com',
+            'new_email2' => 'email@example.com',
+        ];
+
+        $this->actingAs($this->account)->post(route('mship.manage.email.add.post'), $data)
+            ->assertRedirect(route('mship.manage.dashboard'))
+            ->assertSessionHas('success');
+    }
+
+    /** @test * */
+    public function testDuplicateEmailOnPost()
+    {
+        $account = $this->account->secondaryEmails()->create(['email' => 'email2@example.com']);
```

Comment:
Is this required?

Reply:
From what I could gather, this was the only way to test this particular condition within the controller.

Reply:
Seems like it, weird!